SmartDeviceLink/private/SDLTextAndGraphicManager.m

```diff
@@ -150,21 +151,29 @@ - (void)updateWithCompletionHandler:(nullable SDLTextAndGraphicUpdateCompletionH
 
     if (self.isDirty) {
         self.isDirty = NO;
-        [self sdl_updateWithCompletionHandler:handler];
+        [self sdl_updateAndCancelPreviousOperations:YES completionHandler:handler];
     }
 }
 
-- (void)sdl_updateWithCompletionHandler:(nullable SDLTextAndGraphicUpdateCompletionHandler)handler {
+- (void)sdl_updateAndCancelPreviousOperations:(BOOL)supersedePreviousOperations completionHandler:(nullable SDLTextAndGraphicUpdateCompletionHandler)handler {
     SDLLogD(@"Updating text and graphics");
-    if (self.transactionQueue.operationCount > 0) {
+    if (self.transactionQueue.operationCount > 0 && supersedePreviousOperations) {
         SDLLogV(@"Transactions already exist, cancelling them");
         [self.transactionQueue cancelAllOperations];
     }
 
     __weak typeof(self) weakSelf = self;
-    SDLTextAndGraphicUpdateOperation *updateOperation = [[SDLTextAndGraphicUpdateOperation alloc] initWithConnectionManager:self.connectionManager fileManager:self.fileManager currentCapabilities:self.windowCapability currentScreenData:self.currentScreenData newState:[self currentState] currentScreenDataUpdatedHandler:^(SDLShow * _Nonnull newScreenData) {
-        weakSelf.currentScreenData = newScreenData;
-        [weakSelf sdl_updatePendingOperationsWithNewScreenData:newScreenData];
+    SDLTextAndGraphicUpdateOperation *updateOperation = [[SDLTextAndGraphicUpdateOperation alloc] initWithConnectionManager:self.connectionManager fileManager:self.fileManager currentCapabilities:self.windowCapability currentScreenData:self.currentScreenData newState:[self currentState] currentScreenDataUpdatedHandler:^(SDLTextAndGraphicState *_Nullable newScreenData, NSError *_Nullable error) {
+        __strong typeof(weakSelf) strongSelf = weakSelf;
+        if (newScreenData != nil) {
+            // Update our current screen data
+            strongSelf.currentScreenData = newScreenData;
+            [strongSelf sdl_updatePendingOperationsWithNewScreenData:newScreenData];
+        } else if (error != nil) {
+            // Invalidate data that's different from our current screen data
+            [strongSelf sdl_resetFieldsToCurrentScreenData];
```

Comment:
Let's say that a Show fails because of bad data (perhaps unsupported characters were used) and the developer didn't set or didn't check a completion handler, then until whatever text field had that bad character is changed, _every_ Show attempt from then on will fail. Resetting them to the last known "success" state (e.g. what's actually on screen) will fix that issue.

I added a change to not reset these if it's an update superseded failure instead of an actual `Show` failure.